counting/counting_test.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright (c) 2024 Diagrid Inc.
+Licensed under the MIT License.
+*/
+
+package counting
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/diagridio/go-etcd-cron/partitioning"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	etcdclientv3 "go.etcd.io/etcd/client/v3"
+)
+
+const defaultEtcdEndpoint = "127.0.0.1:2379"
+
+func TestCounterTTL(t *testing.T) {
+	ctx := context.TODO()
+	organizer := partitioning.NewOrganizer(randomNamespace(), partitioning.NoPartitioning())
+	etcdClient, err := etcdclientv3.New(etcdclientv3.Config{
+		Endpoints: []string{defaultEtcdEndpoint},
+	})
+	require.NoError(t, err)
+
+	key := organizer.CounterPath(0, "count")
+	// This counter will expire keys 1s after their next scheduled trigger.
+	counter := NewEtcdCounter(etcdClient, key, time.Second)
+
+	value, updated, err := counter.Add(ctx, 1, time.Now().Add(time.Second))
+	require.NoError(t, err)
+	assert.True(t, updated)
+	assert.Equal(t, 1, value)
+
+	value, updated, err = counter.Add(ctx, 2, time.Now().Add(time.Second))
+	require.NoError(t, err)
+	assert.True(t, updated)
+	assert.Equal(t, 3, value)
+
+	time.Sleep(time.Second)
```

Comment:
Please, use the `k8s Clock` so that it can be mocked in tests and can execute immediately. https://github.com/dapr/dapr/blob/ed56f692122d0afef7c79147f77eb283acb10467/pkg/placement/placement.go#L154

Reply:
Can't do it because the expiration is handled by etcd, so we need to wait for a real expiration in the database.